File: src/main/java/com/fendyk/clients/redis/RedisActivities.java
package com.fendyk.clients.redis;

import com.fendyk.DTOs.ActivitiesDTO;
import com.fendyk.DTOs.LandDTO;
import com.fendyk.Main;
import com.fendyk.clients.RedisAPI;
import io.lettuce.core.RedisClient;
import io.lettuce.core.pubsub.RedisPubSubListener;

import javax.xml.stream.events.DTD;
import java.util.ArrayList;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

public class RedisActivities extends RedisAPI<ActivitiesDTO> {
    public RedisActivities(String redisKey) {
        super(redisKey, ActivitiesDTO.class);
    }
}

File: src/main/java/com/fendyk/clients/redis/RedisActivities.kt
package com.fendyk.clients.redis

import com.fendyk.DTOs.ActivitiesDTO
import com.fendyk.clients.RedisAPI
import java.util.UUID
import java.util.concurrent.CompletableFuture

class RedisActivities : RedisAPI<UUID, ActivitiesDTO>("activities:", ActivitiesDTO::class.java) {
    override fun get(key: UUID): CompletableFuture<ActivitiesDTO> {
        return fetch(key.toString())
    }

    override fun set(key: UUID, dto: ActivitiesDTO): CompletableFuture<Boolean> {
        return save(key.toString(), dto)
    }
}
